scylladb/sphinx-scylladb-theme
# Enable banner via Google Tag Manager

Adds the option to enable the banner via Google Tag Manager.

## How to test this pull request

To enable the banner, follow these steps to configure the tag via Google Tag Manager:

1. Log in to Google Tag Manager at https://tagmanager.google.com/.
2. Open the account assigned to this project: `GTM-T8P2JP`.
3. Select **Tags** from the left sidebar.

    ![image](https://user-images.githubusercontent.com/9107969/221174180-6b825c1c-08c7-4dcc-b395-e9728dbfbc78.png)

4. Click **New** and configure the tag with the following settings:

    ![image](https://user-images.githubusercontent.com/9107969/221174271-66902bc6-adf6-4bfe-ad69-957a83da4821.png)

    * **Type:** Custom HTML
   * **Body**:
   
      ```
     <meta name="scylladb-docs-hide_banner" content="false" />
     <meta name="scylladb-docs-banner_title_text" content="Title text" />
     <meta name="scylladb-docs-banner_button_text" content="Button text" />
     <meta name="scylladb-docs-banner_button_url" content="https://docs.scylladb.com" />
      ```
   
   * **Trigger**: On initialization on all pages.
   
5. Click **Save**.

6. Click **Submit** and fill in the information to release a new version of the tag.

    ![image](https://user-images.githubusercontent.com/9107969/221175345-f027ef59-a18b-4562-a657-fddcfd220625.png)

7. Click **Publish**.

8. Clone this pull request, navigate to the docs folder, and run make preview. Open http://127.0.0.1:5500/ in incognito mode to ensure that your browser does not keep any cookies from previous browsing sessions.

You should now see the announcement banner on the website 🎉

**IMPORTANT:** Once you've done the test, remember to delete the tag so that a banner with a sample text appears on all sites once we release a new version of the theme.

## Editing the banner content

You can customize the banner's title, button text, and URL by editing the `content` attribute of the corresponding meta `key`. For example:

```
<meta name="scylladb-docs-hide_banner" content="false" />
<meta name="scylladb-docs-banner_title_text" content="Title text" />
<meta name="scylladb-docs-banner_button_text" content="Button text" />
<meta name="scylladb-docs-banner_button_url" content="https://docs.scylladb.com" />
```

To change a property, edit the tag's body accordingly. Once you've made the changes, remember to publish a new version of the tag.


## Disabling the banner

To disable the banner, you have two options:

a. Delete the tag on Google Tag Manager.

b. Set the meta attribute `scylladb-docs-hide_banner` to `true`.

  ```
     <meta name="scylladb-docs-hide_banner" content="false" />
  ```

In both cases, remember to publish a new version.
